plugins/hls-hlint-plugin/src/Ide/Plugin/Hlint.hs

```diff
@@ -258,15 +265,40 @@ instance Binary GetHlintSettings
 
 type instance RuleResult GetHlintSettings = (ParseFlags, [Classify], Hint)
 
-getHlintSettingsRule :: HlintUsage -> Rules ()
+getHlintSettingsRule :: HlintUsage -> Action (ParseFlags, [Classify], Hint)
 getHlintSettingsRule usage =
-    defineNoFile $ \GetHlintSettings ->
       liftIO $ case usage of
           HlintEnabled cmdArgs -> argsSettings cmdArgs
           HlintDisabled        -> fail "hlint configuration unspecified"
 
 -- ---------------------------------------------------------------------
 
+newtype Config = Config [String]
+
+properties :: Properties '[ 'PropertyKey "flags" ('TArray String)]
+properties = emptyProperties
+  & defineArrayProperty #flags
+    "Flags used by hlint" []
+
+-- | Returns the value of a property for use with shake Rules
+usePropertyAction ::
+  (HasProperty s k t r) =>
+  KeyNameProxy s ->
+  PluginId ->
+  Properties r ->
+  Action (ToHsType t)
+usePropertyAction kn pId p = do
+  config <- getClientConfigAction def
+  let pluginConfig = configForPlugin config pId
+  return $ useProperty kn p $ Just (plcConfig pluginConfig)
```

Comment:
This function is already in ghcide:
https://github.com/haskell/haskell-language-server/blob/0c5a317560c4ee7257ddf03de117868cb7a056c2/ghcide/src/Development/IDE/Core/Rules.hs#L949-L958